projects/ngx-drag-to-select/src/lib/select-container.component.ts

```diff
@@ -382,6 +392,29 @@ export class SelectContainerComponent implements AfterViewInit, OnDestroy {
     });
   }
 
+  private _initSelectionOutputs(mousedown$: Observable<MouseEvent>, mouseup$: Observable<MouseEvent>) {
+    mousedown$
```

Comment:
Not saying you have to change this, but I personally am starting a crusade against global variables 😂 
So the "refactored" version would be:


```ts
mousedown$
      .pipe(
        filter(event => this._cursorWithinHost(event)),
        tap(() => this.selectionStarted.emit()),
        concatMap(() => mouseup$.pipe(first())),
        withLatestFrom(this._selectedItems$),
        map(([, items]) => items),
        takeUntil(this.destroy$)
      )
      .subscribe(items => {
        this.selectionEnded.emit(items);
      });
``` 	

Or a slight variant:

```ts
mousedown$
      .pipe(
        filter(event => this._cursorWithinHost(event)),
        tap(() => this.selectionStarted.emit()),
        concatMap(() =>
          mouseup$.pipe(
            withLatestFrom(this._selectedItems$),
            tap(([, items]) => this.selectionEnded.emit(items)),
            first()
          )
        ),
        takeUntil(this.destroy$)
      )
      .subscribe();
```

Reply:
Yep this would also work. But that is also quite more complex wouldn't you agree? Sometimes I prefer the easy way, which would be to have this tiny flag, instead of more code and more building blocks to understand. But I get your point. Why do you think globals are bad here? I am just super curious to hear your thoughts.

Reply:
But I really like this approach because it only introduces one more stream!

Reply:
I will refactor the code to what you suggested.

Reply:
At first sight it's maybe more complex if you're not fully into RxJS. 
Imho this would be easier to read when you come back to it, also because you don't have to check if the global variable is used/setted in another part of the application.

I think you can't remove all of the global variables, but I'm trying to not introduce them. Mainly because I want to force myself to think reactive and more "pure".

Reply:
Yep good point. Fully agree here.